R/SoilTaxonomyDendrogram.R

```diff
@@ -44,64 +45,78 @@
+#'   OSDexamples$SPC[1:8, ], width = 0.3, name.style = 'center-center',
+#'   KST.order = TRUE
+#' )
 #' 
 #' # classic ordering, based on nominal scale variables (unordered factors)
 #' SoilTaxonomyDendrogram(
-#' OSDexamples$SPC[1:8, ], width = 0.3, name.style = 'center-center',
-#' KST.order = FALSE
+#'   OSDexamples$SPC[1:8, ], width = 0.3, name.style = 'center-center',
+#'   KST.order = FALSE
 #' )
 #' 
 #' 
-SoilTaxonomyDendrogram <- function(spc, KST.order = TRUE, rotationOrder = NULL, name = 'hzname', name.style = 'center-center', id.style = 'side', max.depth = max(spc), n.depth.ticks = 6, scaling.factor = 0.015, cex.names = 0.75, cex.id = 0.75, axis.line.offset = -4, width = 0.1, y.offset = 0.5, shrink = FALSE, font.id = 2, cex.taxon.labels = 0.66, dend.color = par('fg'), dend.width = 1, ...) {
+SoilTaxonomyDendrogram <- function(spc, KST.order = TRUE, rotationOrder = NULL, level = c("soilorder", "suborder", "greatgroup", "subgroup"), name = 'hzname', name.style = 'center-center', id.style = 'side', max.depth = max(spc), n.depth.ticks = 6, scaling.factor = 0.015, cex.names = 0.75, cex.id = 0.75, axis.line.offset = -4, width = 0.1, y.offset = 0.5, shrink = FALSE, font.id = 2, cex.taxon.labels = 0.66, dend.color = par('fg'), dend.width = 1, ...) {
 	
   
   # attempt KST-based ordering:
   # 1. setup ordinal factors based on order of taxa with each level of ST hierarchy
   # 2. rotate dendrogram to reflect ordering of subgroups within keys
-  # note: will create NA if obsolete taxa or typos -> test for this 
-  if(KST.order) {
+  # note: will create NA if obsolete taxa or typos -> test for this
+  if (KST.order) {
+    
     
     # requires SoilTaxonomy >= 0.1.5 (2022-02-15)
-    if(!requireNamespace('SoilTaxonomy', quietly=TRUE)) {
-      stop('please install the `SoilTaxonomy` packages', call.=FALSE)
+    if (!requireNamespace('SoilTaxonomy', quietly = TRUE)) {
+      stop('please install the `SoilTaxonomy` packages', call. = FALSE)
     }
     
-    # CRAN check hack
+    # TODO: a function to set ordered factors in a data.frame-like object should be added to SoilTaxonomy
     ST_unique_list <- NULL
-      
+    
     # note: this is incompatible with LazyData: true
-    load(system.file("data/ST_unique_list.rda", package="SoilTaxonomy")[1])
+    load(system.file("data/ST_unique_list.rda", package = "SoilTaxonomy")[1])
     
-    # create ordered factors, dropping unused levels
-    .soilorder <- droplevels(factor(spc$soilorder, levels = ST_unique_list$order, ordered = TRUE))
-    .suborder <- droplevels(factor(spc$suborder, levels = ST_unique_list$suborder, ordered = TRUE))
-    .greatgroup <- droplevels(factor(spc$greatgroup, levels = ST_unique_list$greatgroup, ordered = TRUE))
-    .subgroup <- droplevels(factor(spc$subgroup, levels = ST_unique_list$subgroup, ordered = TRUE))
+    # support for NASIS physical column names (NASIS possibly should be default?)
+    if (is.null(spc$soilorder) & !is.null(spc$taxorder)) {
```

Comment:
> No need to check for specific names. 

The suggestion (top-down ordering for non-ordered classes specified to `level`) would be fine for the `KST.order=FALSE` case. I think that in order to set the _ordered_ factor levels as we do with default `KST.order=TRUE` there needs to be some linkage between sets of _possible_ names and the corresponding levels that would get pulled from `ST_unique_list`.

I agree it could be more generic than currently implemented--but covering the standard NASIS columns is really just the bare minimum that we should expect for functions like this--and that was the only goal for this PR. 

So some specificity, or at least a mechanism to do the mapping via argument, is required for the default case.

`level` could be a _named_ character, where names are standard US hierarchy positions and the values refer to the column name used for lookup. e.g. `c(greatgroup="taxgrtgroup")`. More generally, we probably should have a way to handle other ordered levels not defined in {SoilTaxonomy} (obsolete/new taxonomy; non-US taxonomic systems)